src/libraries/Microsoft.Extensions.Configuration.Binder/src/ConfigurationBinder.cs

```diff
@@ -344,77 +343,72 @@ private static void BindProperty(PropertyInfo property, object instance, IConfig
                 }
 
                 // Leaf nodes are always reinitialized
-                return convertedValue;
+                bindingPoint.TrySetValue(convertedValue);
+                return;
             }
 
             if (config != null && config.GetChildren().Any())
             {
+                // for arrays and read-only list-like interfaces, we concatenate on to what is already there
+                if (type.IsArray || IsArrayCompatibleReadOnlyInterface(type))
```

Comment:
Prior to this change, we'd assign `List<T>`s to settable `IEnumerable<T>`, `IReadOnlyCollection<T>` and `IReadOnlyList<T>` properties. Now these will be arrays. I think this is okay, just pointing it out.

Reply:
@halter73 good callout; I should have left a comment about this. I made the change just to reduce some otherwise-duplicative code. As you say, I think it should be ok. Hopefully noone is assuming they can cast one of the read-only interfaces to a specific concrete implementation.